security/filter/security-filter-login/src/main/java/org/codice/ddf/security/filter/login/LoginFilter.java

```diff
@@ -253,11 +257,8 @@ private Subject handleAuthenticationToken(HttpServletRequest httpRequest, SAMLAu
                             createStatus(SAMLProtocolResponseValidator.SAML2_STATUSCODE_SUCCESS,
                                     null));
 
-                    if (docBuilder == null) {
-                        throw new SecurityServiceException("Unable to validate SAML assertions.");
-                    }
-
-                    Document doc = docBuilder.newDocument();
+                    BUILDER.get().reset();
```

Comment:
This is done to avoid having to create a new document builder.

"DocumentBuilder is reset to the same state as when it was created with DocumentBuilderFactory.newDocumentBuilder(). reset() is designed to allow the reuse of existing DocumentBuilders thus saving resources associated with the creation of new DocumentBuilders."

http://docs.oracle.com/javase/7/docs/api/javax/xml/parsers/DocumentBuilder.html#reset()